### ISSUE TITLE
suggestion Bukkit.removeRecipe(key)

### ISSUE THREAD

**Issue by Anarchick** [ShaneBeee/SkBee#6]
Since 1.15 , bukkitAPI add a method to remove specific recipe via his key, Bukkit.removeRecipe(key). It's maybe more powerfull than loop through all recipes and search the matching key.

also, you use this : 
`recipeMap.values().forEach(recipes -> recipes.entrySet().removeIf(entry -> entry.getKey().equals(key)))`

It's a good 1 line of code but, even if I'm not a java dev, I think that you should add a break and not iterate ALL RECIPES
https://stackoverflow.com/questions/23308193/break-or-return-from-java-8-stream-foreach
it's just a suggestion of optimisation :)

**Comment by ShaneBeee** [comment]
> Since 1.15 , bukkitAPI add a method to remove specific recipe via his key, Bukkit.removeRecipe(key). It's maybe more powerfull than loop through all recipes and search the matching key.

Yeah, Im aware of this:
<img width="823" alt="Screen Shot 2020-04-10 at 12 53 12 AM" src="https://user-images.githubusercontent.com/20008482/78973950-ba3e1c00-7ac5-11ea-8fd8-e825488f1ab1.png">
I added it to Bukkit ;)
Due to this being added in 1.15.2, it won't be available for 1.13->1.15.1, therefor there is really no point using it in this plugin.

> also, you use this :
`recipeMap.values().forEach(recipes -> recipes.entrySet().removeIf(entry -> entry.getKey().equals(key)))`

This is basically the exact same way I did it in Bukkit. In Bukkit I was able to do it a bit easier, but due to avoiding using NMS I have to do it that way, so I can filter and use as little NMS as possible.

> It's maybe more powerfull than loop through all recipes and search the matching key.

Before saying this, I recommend looking how its actually done in Bukkit/CB, because it also loops:
https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/0098037647cf474aaf1653b5e9ea90e7e9f16580

Also:
> it's just a suggestion of optimisation :)

Can you please show proof of how the method I am using is not optimistic?
Based on:
> even if I'm not a java dev, 

It appears you dont really know java, and understand how all this works.

Long story short, Minecraft handles recipes within a map of maps, its confusing and difficult to deal with without iterating thru all the values.

**Comment by Anarchick** [comment]
Ok, I had search the source code of Bukkit.removeRecipe() but I didn't find it (not a java dev like a said ^^) , good to see that it's you who add this method !

For the second point :
I think this code : 
`recipeMap.values().forEach(recipes -> recipes.entrySet().removeIf(entry -> entry.getKey().equals(key)))`
do something like this:
```
loop {_recipes::*}:
    if loop-value is {_key}:
        remove loop-value from {_recipes::*}
```

I was suggesting to add a break if the recipe is removed :
```
loop {_recipes::*}:
    if loop-value is {_key}:
        remove loop-value from {_recipes::*}
        stop
```
(yes I know this skript is dumb, but it's for the logical of how I undersand your java code)
I think it's more optmized cause theyre a lot of entries and more at each MC update

Ps: I am not criticizing your code, I know that you are very competent.

**Comment by ShaneBeee** [comment]
I can't change the code, because Im using reflection in the code. 

Also, the way you are doing it with the stop, is exactly what the java `recipes.entrySet().removeIf(entry -> ` does. it removes if it matches.
